examples/DemoFile.Example.WebAssembly/Program.cs

```diff
@@ -0,0 +1,118 @@
+﻿using System.Diagnostics;
+using System.Runtime.InteropServices.JavaScript;
+using System.Text;
+
+namespace DemoFile.Example.WebAssembly;
+
+public partial class Program
+{
+    public static async Task Main(string[] args)
+    {
+        Console.WriteLine("Main()");
+    }
+
+    [JSExport]
+    public static async Task ParseToEndFromUrl(string url)
+    {
+        Console.WriteLine($"Parsing demo from url: {url}");
+        using var client = new HttpClient();
+        using var stream = await client.GetStreamAsync(url);
+        await ParseToEndFromStream(stream);
+    }
+
+    [JSExport]
+    public static async Task ParseToEndFromArray(byte[] buffer)
+    {
+        using var stream = new MemoryStream(buffer);
+        await ParseToEndFromStream(stream);
+    }
+
+    private static async Task ParseToEndFromStream(Stream stream)
+    {
+        Console.WriteLine($"Parsing started, stream length {stream.Length}");
+
+        SetDemoParseResult(string.Empty);
+        SetDemoParseProgress(0);
+
+        var demo = new DemoParser();
+        var sb = new StringBuilder(1024);
+
+        var roundNum = 0;
+        demo.Source1GameEvents.RoundStart += e =>
+        {
+            roundNum += 1;
+            sb.Append("<br />");
+            sb.Append("<br />");
+            sb.AppendLine($"\n\n>>> Round start [{roundNum}] <<<");
+            sb.Append("<br />");
+        };
+
+        demo.Source1GameEvents.RoundFreezeEnd += e =>
+        {
+            sb.Append("<br />");
+            sb.AppendLine("\n  > Round freeze end");
+            sb.Append("<br />");
+        };
+
+        demo.Source1GameEvents.RoundEnd += e =>
+        {
+            sb.Append("<br />");
+            sb.AppendLine("\n  > Round end");
+            sb.Append("<br />");
+        };
+
+        demo.Source1GameEvents.PlayerDeath += e =>
+        {
+            sb.AppendLine($"{e.Attacker?.PlayerName} [{e.Weapon}] -> {e.Player?.PlayerName}");
+            sb.Append("<br />");
+        };
+
+        var totalTimeStopwatch = Stopwatch.StartNew();
+        var delayStopwatch = Stopwatch.StartNew();
+        var processingTimeStopwatch = Stopwatch.StartNew();
+
+        await demo.StartReadingAsync(stream, default);
+
+        while (await demo.MoveNextAsync(default))
```

Comment:
I think this should be driven by the web page, using something like `setTimeout`. The webpage should call `MoveNextAsync` in a loop until 16.67ms (for 60fps) or 31.25ms (for 30fps) have passed. This should keep the UI completely unblocked, but parsing should still progress at a reasonable speed.

Reply:
Yes, it can be done with `setTimeout`. But it will not make things better. The parser will still block the UI for 16.67ms. There is no background thread which can continue execution of `MoveNextAsync()`. And 16.67ms is a big overhead, like I said, I had to reduce it to 20 FPS (50 ms).

Reply:
The only proper way to unblock browser thread, is to use [Web Workers](https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/Using_web_workers), which is out of the scope of this PR. This PR is meant to provide simple example how the library can be used on Web. If you want, we can later on add this functionality.